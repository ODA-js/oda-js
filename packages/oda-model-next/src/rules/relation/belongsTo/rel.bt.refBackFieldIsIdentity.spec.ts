import 'jest';
import Rule from './refBackFieldIsIdentity';

describe('rule', () => {
  let rule: Rule;
  beforeAll(() => {
    rule = new Rule();
  });

  it('throw', () => {
    const result = rule.validate({
      relation: {
        ref: {
          backField: true
        }
      },
      entity: {
        fields: {
          get: jest.fn(() => ({
            identity: true
          }))
        }
      }
    } as any);
    expect(result).toMatchSnapshot();
  });

  it('not throw 1', () => {
    const result = rule.validate({
      relation: {
        ref: {
          backField: true
        }
      },
      entity: {
        fields: {
          get: jest.fn(() => ({
            identity: false
          }))
        }
      }
    } as any);
    expect(result).toMatchSnapshot();
  });

  it('not throw 2', () => {
    const result = rule.validate({
      relation: {
        ref: {
          backField: true
        }
      },
      entity: {
        fields: {
          get: jest.fn(() => null)
        }
      }
    } as any);
    expect(result).toMatchSnapshot();
  });
});
