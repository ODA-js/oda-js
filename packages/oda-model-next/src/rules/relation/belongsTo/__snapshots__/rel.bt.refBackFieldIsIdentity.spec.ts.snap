// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule not throw 1 1`] = `Array []`;

exports[`rule not throw 2 1`] = `Array []`;

exports[`rule throw 1`] = `
Array [
  Object {
    "message": "back field is identity",
    "result": "critics",
  },
]
`;
